docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -305,6 +305,50 @@ In the example below:
 If you are dynamically switching the `leafField` or `mainGroupingCriteria`, the sorting and filtering models will not be cleaned up automatically, and the sorting/filtering will not be re-applied.
 :::
 
+## Automatic parents and children selection
+
+By default, selecting a parent row does not select its children.
+You can override this behavior by using the `rowSelectionPropagation` prop.
+
+Here's how it's structured:
+
+```ts
+type GridRowSelectionPropagation = {
+  descendants?: boolean; // default: false
+  parents?: boolean; // default: false
+};
+```
+
+When `rowSelectionPropagation.descendants` is set to `true`.
+
+- Selecting a parent would auto-select all its filtered descendants.
+- Deselecting a parent row would auto-deselect all its filtered descendants.
+
+When `rowSelectionPropagation.parents` is set to `true`.
+
+- Selecting all the filtered descendants of a parent would auto-select the parent.
+- Deselecting a descendant of a selected parent would auto-deselect the parent.
+
+The example below demonstrates the usage of the `rowSelectionPropagation` prop.
+
+{{"demo": "RowGroupingPropagateSelection.js", "bg": "inline", "defaultCodeOpen": false}}
```

Comment:
Is this an expected behavior?

https://github.com/user-attachments/assets/457912da-9bcc-42de-9b20-0f95d38cc0d1

Reply:
Not exactly, `id=0` is failing on `!variable` check.
Fixed: [`c16e7f4` (#13757)](https://github.com/mui/mui-x/pull/13757/commits/c16e7f4b8d60f1a59729555ee8cdad12a0a06eb2)